examples/Localization/localization_example_lorenz96.jl

```diff
@@ -0,0 +1,130 @@
+using Distributions
+using LinearAlgebra
+using Random
+
+using Plots
+
+using EnsembleKalmanProcesses
+using EnsembleKalmanProcesses.ParameterDistributions
+using EnsembleKalmanProcesses.Localizers
+import EnsembleKalmanProcesses: construct_mean, construct_cov, construct_sigma_ensemble
+const EKP = EnsembleKalmanProcesses
+
+function rk4(f::Function, y0::Array{Float64, 1}, t0::Float64, t1::Float64, h::Float64; inplace::Bool = true)
+    y = y0
+    n = round(Int, (t1 - t0) / h)
+    t = t0
+    if ~inplace
+        hist = zeros(n, length(y0))
+    end
+    for i in 1:n
+        k1 = h * f(t, y)
+        k2 = h * f(t + 0.5 * h, y + 0.5 * k1)
+        k3 = h * f(t + 0.5 * h, y + 0.5 * k2)
+        k4 = h * f(t + h, y + k3)
+        y = y + (k1 + 2 * k2 + 2 * k3 + k4) / 6
+        if ~inplace
+            hist[i, :] = y
+        end
+        t = t0 + i * h
+    end
+    if ~inplace
+        return hist
+    else
+        return y
+    end
+end
+
+function lorenz96(t, u, p)
+    N = p["N"]
+    F = 8
+
+    du = similar(u)
+
+    for i in 1:N
+        du[i] = (u[mod(i + 1, 1:N)] - u[mod(i - 2, 1:N)]) * u[mod(i - 1, 1:N)] - u[i] + F
+    end
+
+    return copy(du)
+end
+
+D = 200
+
+lorenz96_sys = (t, u) -> lorenz96(t, u, Dict("N" => D))
+
+# Seed for pseudo-random number generator
+rng_seed = 42
+rng = Random.MersenneTwister(rng_seed)
+
+dt = 0.05
+y0 = rk4(lorenz96_sys, randn(D), 0.0, 1000.0, dt)
+
+# Lorenz96 initial condition problem - Section 6.3 of Tong and Morzfeld (2022)
+G(u) = mapslices((u) -> rk4(lorenz96_sys, u, 0.0, 0.4, dt), u, dims = 1)
+N_ens = 20
+p = D
+N_iter = 20
+# Generate random truth
+y = y0 + randn(D)
+Γ = 1.0 * I
+
+#### Define prior information on parameters
+priors = ParameterDistribution[] #empty PD-array
+for i in 1:p
+    push!(priors, ParameterDistribution(Parameterized(Normal(0.0, 1)), no_constraint(), string("u", i)))
+end
```

Comment:
```suggestion
priors = map(1:p) do i
    ParameterDistribution(Parameterized(Normal(0.0, 1)), no_constraint(), string("u", i))
end
```
`push!` is good to use when you don't know how large the collection may be.